src/core/public/chrome/nav_controls/nav_controls_service.ts

```diff
@@ -104,7 +116,17 @@ export class NavControlsService {
         navControlsExpandedCenter$.next(
           new Set([...navControlsExpandedCenter$.value.values(), navControl])
         ),
-
+      registerRightNavigation: (props: RightNavigationProps) => {
+        const nav = {
+          order: props.order,
+          mount: mountReactNode(
+            React.createElement(RightNavigationButton, {
+              ...props,
+            })
```

Comment:
```suggestion
            React.createElement(RightNavigationButton, props)
```
Seems the spread is not needed here.